62.CAD/main.cpp

```diff
@@ -234,52 +407,866 @@ class CPolyline
-
-		2. Implicitization
-			See Chapter 17.6 of https://scholarsarchive.byu.edu/cgi/viewcontent.cgi?article=1000&context=facpub
-			You need to implicitize the quadratic bezier curve which results in a polynomial like this: ax^2+by^2+cxy+dx+ey+f.
-			that's beautiful cause all you need to store this is 6 doubles just like a quadratic bezier,
-			you could even standardize and divide every component by 'a' and use 5 doubles, but let's not do that yet, I'm a bit scared of divisions and haven't thought about this fully
-			We need a implicitized curve per curve (1 to 1 mapping) in our algorithm, we don't need to store these in the Hatch class
-			And here is my desmos showing the implicitization process https://www.desmos.com/calculator/8jfbzqrazh
-
-		3. for the segment sorting you also need to evaluate derivatives in the case that multiple beziers go through the same point
-			(Talk with Matt, he figure out the math)
-	*/
-
+public:
 	// this struct will be filled in cpu and sent to gpu for processing as a single DrawObj
-	struct CurveBox
+	struct CurveHatchBox
 	{
-		// aabb (float64_t2 min, max)
-		// reference to min curve (could be left curve if sweeping in y dir) and it's tmin tmax
-		// reference to max curve (could be right curve if sweeping in y dir) and it's tmin tmax
-		// any reference to texture or colour or style used to fill it Or we could fill that in It's drawObj (latter is better if we could alias with lineStyle address)
+		float64_t2 aabbMin, aabbMax;
+		float64_t2 curveMin[3];
+		float64_t2 curveMax[3];
 	};
 
-	// note: even though in this example we can reference to the polyline bezier and lines somehow, we want to eventualy be able to serialize/deserialize 
-	// this object and should be independant of any outside references so here we will also be keeping a list/vector of quadratic beziers which CurveBox can index into
-	// we have two different types (line,bezier) but we don't want to keep two seperate lists, we will have the lines have the mid point (p1) set to nan adn everything as "beziers"
+	// TODO: start using A, B, C here
+	struct QuadraticBezier {
+		float64_t2 p[3];
+
+		std::array<double, 4> linePossibleIntersections(const QuadraticBezier& other) const;
+		double intersectOrtho(double coordinate, int major) const;
+		float64_t2 evaluateBezier(double t) const;
+		float64_t2 tangent(double t) const;
+		std::array<double, 4> getRoots() const;
+		QuadraticBezier splitCurveTakeLeft(double t) const;
+		QuadraticBezier splitCurveTakeRight(double t) const;
+		// Splits the bezier into monotonic segments. If it already was monotonic, 
+		// returns a copy of this bezier in the first value of the array
+		std::array<QuadraticBezier, 2> splitIntoMonotonicSegments(int major) const;
+		std::pair<float64_t2, float64_t2> getBezierBoundingBoxMinor(int major) const;
```

Comment:
use the constexpr `SelectedMajor` in common.hlsl don't take as input, also make a constexpr  SelectedMinor = 1 -SelectedMajor